django/db/models/fields/related_descriptors.py

```diff
@@ -1130,10 +1105,10 @@ def _add_items(self, source_field_name, target_field_name, *objs):
 
                     # Add the ones that aren't there already
                     self.through._default_manager.using(db).bulk_create([
-                        self.through(**{
+                        self.through(**dict(through_defaults, **{
                             '%s_id' % source_field_name: self.related_val[0],
                             '%s_id' % target_field_name: obj_id,
```

Comment:
Also the `%s_id` is probably going to break for fields that define different `db_column`s. This wasn't an issue before because this branch could only be reached with auto-created intemediary models but now that we allow all forms of `through` we should use `attname` or directly `*_name` here.

Reply:
My assumptions here seems to be wrong, this should work fine.

https://github.com/django/django/blob/d212bc03bac1bce074207074a45032bb6bfdf3ec/django/db/models/fields/related.py#L916-L917